core/amber/src/main/scala/edu/uci/ics/texera/web/service/WorkflowService.scala

```diff
@@ -100,7 +100,7 @@ class WorkflowService(
         (oldState, newState) =>
           {
             if (oldState.state != COMPLETED && newState.state == COMPLETED) {
-              lastCompletedLogicalPlan = Option.apply(executionService.workflow.originalLogicalPlan)
+              lastCompletedLogicalPlan = Option.apply(executionService.workflow.logicalPlan)
             }
             Iterable.empty
```

Comment:
do we still need this diff handler?

Reply:
We do need it. Because `EditingTimeCompilationRequest` needs the `lastCompletedLogicalPlan`. I think the next PR will remove the `EditingTimeCompilationRequest` along with `lastCompletedLogicalPlan` and diff handler.